### DISCUSSION TITLE
Comments break the flow of the code - add support to collapse / hide comments

### DISCUSSION THREAD

**Discussion by redkhalil** [community/community#47210] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

After making comments, the comments remain displayed and this breaks the flow of the code reading. A suggestion is that we should be able to collapse / hide the comment, and have an icon with a number (e.g. like a chat bubble icon) to indicate if this line has any comments. Even better, hovering over that icon should pop-up the comment.

![image](https://user-images.githubusercontent.com/16011713/218389971-94ce5157-b910-4f45-a8b5-867674951c5f.png)

**Comment by andreahlert** [comment]
Hi there 🙂
Yes, long threads can really break concentration when you’re trying to read the diff. A stronger collapse or “focus mode” sounds useful.

Maybe the deeper question is whether users want temporary visual hiding or true conversation minimization logic. The current “hide resolved” helps, but it doesn’t fully address high-volume discussions where even unresolved comments are just context, not blockers.

**Comment by abeehler** [comment]
This is a feature of ADO that was very useful.  Made reading the code for code review much easier.  I have to mentally try to block out the comment blocks in the middle of code.  They are major distractions and break flow of code reading.
